singer_sdk/streams/sql.py

```diff
@@ -825,9 +830,33 @@ def _adapt_column_type(
         Raises:
             NotImplementedError: if altering columns is not supported.
         """
-        current_type = self._get_column_type(full_table_name, column_name)
+        current_type: sqlalchemy.types.TypeEngine = self._get_column_type(
+            full_table_name, column_name
+        )
+
+        # Check if the existing column type and the sql type are the same
+        if isinstance(sql_type, type(current_type)):
+            # The current column and sql type are the same
+            # Nothing to do
+            return
+        # Check if the existing column generic type
+        # and sql generic type are the same
+        elif isinstance(sql_type.as_generic(), type(current_type.as_generic())):
+            # The current column and sql generic types are the same
+            # Nothing to do
+            return
+        # Check if the new type is a subclass of the current type
+        elif issubclass(type(sql_type.as_generic()), type(current_type)):
+            # Since it is a sub class of the current type
+            # a conversion on insert will happen
+            # Nothing to do
+            return
+
+        # Not the same type, generic type or compatible types
+        # calling merge_sql_types for assistnace
         compatible_sql_type = self.merge_sql_types([current_type, sql_type])
-        if current_type == compatible_sql_type:
+
+        if compatible_sql_type == current_type:
```

Comment:
I don't understand SQL Alchemies `__eq__` override on Types enough yet for this.

Reply:
If you are asking why I swapped them.  It was purely to keep things on the same side.  I had been putting the `current_type` on the righthand side above and just wanted to keep the visual continuity.

Reply:
I switched this to a str() to st() evaluation to keep consistent.  The added benefit is I believe we are steersing clear of the `__eq__` override that sqlalchemy has in place.